pkg/cloud-init/cloud-init.go

```diff
@@ -162,15 +163,73 @@ func ReadCloudInitVolumeDataSource(vmi *v1.VirtualMachineInstance, secretSourceD
 	return nil, nil
 }
 
+func resolveSSHPublicKeys(accessCredentials []v1.AccessCredential, secretSourceDir string, methodNoCloud, methodConfigDrive bool) (map[string]string, error) {
```

Comment:
could you structure the code in such a way we avoid the two boolean args ? Maybe inject the "validation function" from the client.

For clarity, when I mean you could pass the following as an input argument:
```golang
func isNoCloudAccessCredential(accessCredential v1.AccessCredential) bool {
    return accessCred.SSHPublicKey == nil || accessCred.SSHPublicKey.PropagationMethod.NoCloud
}

func isConfigDriveAccessCredential(accessCredential v1.AccessCredential) bool {
    return accessCred.SSHPublicKey == nil || accessCred.SSHPublicKey.PropagationMethod.NoCloud
}

func resolveSSHPublicKeys(accessCredentials []v1.AccessCredential, secretSourceDir string, isAccessCredentialValidFunc func(v1.AccessCredential) bool) (map[string]string, error) { ...}
```